### PR TITLE
support metric vector

### PR DESCRIPTION
**Pull request by siddontang**
@BusyJay @overvenus 

PTAL @lucab

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @BusyJay @lucab

File: src/errors.rs [hunk 1]
```diff
@@ -19,11 +19,10 @@ quick_error!{
     pub enum Error {
         AlreadyReg {
             description("duplicate metrics collector registration attempted")
-            display("duplicate metrics collector registration attempted")
         }
-        InconsistentCardinality {
+        InconsistentCardinality(need: usize, got: usize) {
```
Comment (overvenus) [inline src/errors.rs]: s/need/expect

File: src/vec.rs [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+    /// `with_label_values` works as `get_metric_with_label_values`, but panics if an error
+    /// occurs. The method allows neat syntax like:
+    ///     httpReqs.with_label_values("404", "POST").inc()
+    pub fn with_label_values(&self, vals: &[&str]) -> T::Output {
+        self.get_metric_with_label_values(vals).unwrap()
+    }
+
+    /// `with` works as `get_metric_with`, but panics if an error occurs. The method allows
+    /// neat syntax like:
+    ///     httpReqs.with(Labels{"status":"404", "method":"POST"}).inc()
```
Comment (overvenus) [inline src/vec.rs]: Maybe we should tweak this syntax?

Comment (siddontang) [inline src/vec.rs]: just copy from go comment. 
You can help me to support a label macro to support this syntax later :-)

**Comment by overvenus** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**